Potential NPE in RewriteProxiesEventHandler#rewriteProxyURIs

* [x] **I have checked that this bug has not yet been reported by someone else**
* [x] **I have checked that this bug appears on Chrome**
* [x] **I have specified the version** :
- 2024.5.6
* [x] **I have specified my environment** : 
- All

### Stack

When I try to upload a project in SysON 2024.5.3 (with Sirius Web 2024.5.6) I get the following exception:

```
Caused by: java.lang.NullPointerException: Cannot invoke "org.eclipse.emf.ecore.InternalEObject.eIsProxy()" because "internalEObject" is null
	at org.eclipse.sirius.web.application.project.services.RewriteProxiesEventHandler.lambda$rewriteProxyURIs$0(RewriteProxiesEventHandler.java:84) ~[sirius-web-application-2024.5.6.jar:2024.5.6]
	at java.base/java.lang.Iterable.forEach(Iterable.java:75) ~[na:na]
	at org.eclipse.sirius.web.application.project.services.RewriteProxiesEventHandler.lambda$rewriteProxyURIs$1(RewriteProxiesEventHandler.java:82) ~[sirius-web-application-2024.5.6.jar:2024.5.6]
	at java.base/java.util.Iterator.forEachRemaining(Iterator.java:133) ~[na:na]
	at org.eclipse.sirius.web.application.project.services.RewriteProxiesEventHandler.rewriteProxyURIs(RewriteProxiesEventHandler.java:81) ~[sirius-web-application-2024.5.6.jar:2024.5.6]
	at org.eclipse.sirius.web.application.project.services.RewriteProxiesEventHandler.handle(RewriteProxiesEventHandler.java:67) ~[sirius-web-application-2024.5.6.jar:2024.5.6]
	at org.eclipse.sirius.components.collaborative.editingcontext.EditingContextEventProcessor.handleInput(EditingContextEventProcessor.java:335) ~[sirius-components-collaborative-2024.5.6.jar:2024.5.6]
	at org.eclipse.sirius.components.collaborative.editingcontext.EditingContextEventProcessor.doHandle(EditingContextEventProcessor.java:277) ~[sirius-components-collaborative-2024.5.6.jar:2024.5.6]
	at org.eclipse.sirius.components.collaborative.editingcontext.EditingContextEventProcessor.lambda$handle$2(EditingContextEventProcessor.java:241) ~[sirius-components-collaborative-2024.5.6.jar:2024.5.6]
	at org.eclipse.sirius.components.web.concurrent.DelegatingRequestContextRunnable.run(DelegatingRequestContextRunnable.java:40) ~[sirius-components-web-2024.5.6.jar:2024.5.6]
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539) ~[na:na]
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264) ~[na:na]
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) ~[na:na]
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) ~[na:na]
	... 1 common frames omitted
```

### Steps to reproduce

Try to upload this project:
[Batmobile-Imported.zip](https://github.com/user-attachments/files/16194704/Batmobile-Imported.zip)

May be the project contains bad cross references but I don't think it should be a reason for `RewriteProxiesEventHandler#rewriteProxyURIs` to fail.

--- a/CHANGELOG.adoc
+++ b/CHANGELOG.adoc
@@ -61,6 +61,7 @@ As a result, the following maven modules have been deleted: `sirius-web-sample-a
 - https://github.com/eclipse-sirius/sirius-web/issues/3767[#3767] [diagram] Improves the robustness of the arrange all
 - https://github.com/eclipse-sirius/sirius-web/issues/3784[#3784] [sirius-web] Fix the click on a representation which is already selected in the explorer.
 - https://github.com/eclipse-sirius/sirius-web/issues/2759[#2759] [diagram] Reactivate the Selection Dialog
+- https://github.com/eclipse-sirius/sirius-web/issues/3766[#3766] [sirius-web] Fix a potential NPE in RewriteProxiesEventHandler#rewriteProxyURIs
 
 === New Features
 

--- a/packages/sirius-web/backend/sirius-web-application/src/main/java/org/eclipse/sirius/web/application/project/services/RewriteProxiesEventHandler.java
+++ b/packages/sirius-web/backend/sirius-web-application/src/main/java/org/eclipse/sirius/web/application/project/services/RewriteProxiesEventHandler.java
@@ -81,7 +81,7 @@ public class RewriteProxiesEventHandler implements IEditingContextEventHandler {
         resource.getAllContents().forEachRemaining(eObject -> {
             eObject.eCrossReferences().forEach(target -> {
                 InternalEObject internalEObject = (InternalEObject) target;
-                if (internalEObject.eIsProxy()) {
+                if (internalEObject != null && internalEObject.eIsProxy()) {
                     URI proxyURI = internalEObject.eProxyURI();
                     String oldDocumentId = proxyURI.path().substring(1);
                     String newDocumentId = oldDocumentIdToNewDocumentId.get(oldDocumentId);

